src/wallet/test/init_tests.cpp

```diff
@@ -73,6 +73,8 @@ BOOST_AUTO_TEST_CASE(walletinit_verify_walletdir_no_trailing)
     BOOST_CHECK_EQUAL(walletdir, expected_path);
 }
 
+#ifndef WIN32
+// Windows does not consider "datadir/wallets//" to be a valid directory path.
```

Comment:
In commit "refactor: replace boost::filesystem with std::filesystem" (41d7166c8a598b604aad6c6b1d88ad46e23be247)

What's the behavior if you use this path on windows? I don't think we want bitcoins's behavior undefined and untested in this case. It would be better for users and also clearer in the test code to check what the resulting behavior is on windows and decide if it is good or bad.

Or best thing might be to add a StripRedundantLastElementsOfPath call in the `-walletdir=` parsing code like the one that was added to GetDataDir in this PR, so this test could go back to be enabled and working the same across platforms.

Reply:
I was assuming that this modifies `gArgs`, but given that the args is local, it should be fine, assuming that the argsmanager doesn't write to disk by itself.